components/lib/calendar/Calendar.js

```diff
@@ -1961,7 +1961,7 @@ export const Calendar = React.memo(
 
             let formattedValue = '';
 
-            if (value) {
+            if (value instanceof Date) {
```

Comment:
I don't think you can do this here because `issingleSelection` its a date but `isRangeSelection` and `isMultipleSelection` the value is not a DATE its an array of dates.

Reply:
You are right